core-drawer-panel.html

```diff
@@ -75,344 +75,425 @@
-    },
-    
-    tapHandler: function(e) {
-      if (e.target && this.toggleAttribute && 
-          e.target.hasAttribute(this.toggleAttribute)) {
-        this.togglePanel();
-      }
-    },
+      onQueryMatchesChanged: function(event) {
+        this.narrow = event.detail.value || this.forceNarrow;
 
-    isEdgeTouch: function(e) {
-      return !this.disableEdgeSwipe && this.swipeAllowed() &&
-        (this.rightDrawer ?
-          e.pageX >= this.offsetWidth - this.edgeSwipeSensitivity :
-          e.pageX <= this.edgeSwipeSensitivity);
-    },
+        if (this.narrow) {
+          this.selected = this.defaultSelected;
+        }
 
-    trackStart : function(e) {
-      if (this.swipeAllowed()) {
-        this.dragging = true;
+        this.setAttribute('touch-action', this.swipeAllowed() ? 'pan-y' : '');
+        this.fire('core-responsive-change', {narrow: this.narrow});
+      },
 
-        if (this.isMainSelected()) {
-          this.dragging = this.peeking || this.isEdgeTouch(e);
-        }
+      forceNarrowChanged: function() {
+        this.queryMatchesChanged();
+      },
 
-        if (this.dragging) {
-          this.width = this.$.drawer.offsetWidth;
-          this.transition = false;
-          e.preventTap();
-        }
-      }
-    },
-
-    translateXForDeltaX: function(deltaX) {
-      var isMain = this.isMainSelected();
-      if (this.rightDrawer) {
-        return Math.max(0, isMain ? this.width + deltaX : deltaX);
-      } else {
-        return Math.min(0, isMain ? deltaX - this.width : deltaX);
-      }
-    },
+      swipeAllowed: function() {
+        return this.narrow && !this.disableSwipe;
+      },
 
-    trackx : function(e) {
-      if (this.dragging) {
+      isMainSelected: function() {
+        return this.selected === 'main';
+      },
+
+      startEdgePeek: function() {
+        this.width = this.$.drawer.offsetWidth;
+        this.moveDrawer(this.translateXForDeltaX(this.rightDrawer ?
+            -this.edgeSwipeSensitivity : this.edgeSwipeSensitivity));
+        this.peeking = true;
+      },
+
+      stopEdgePeek: function() {
         if (this.peeking) {
-          if (Math.abs(e.dx) <= this.edgeSwipeSensitivity) {
-            return; // Ignore trackx until we move past the edge peek.
-          }
           this.peeking = false;
+          this.moveDrawer(null);
         }
-        this.moveDrawer(this.translateXForDeltaX(e.dx));
-      }
-    },
+      },
 
-    trackEnd : function(e) {
-      if (this.dragging) {
-        this.dragging = false;
-        this.transition = true;
-        this.moveDrawer(null);
+      downHandler: function(e) {
+        if (!this.dragging && this.isMainSelected() && this.isEdgeTouch(e)) {
+          this.startEdgePeek();
+        }
+      },
+
+      upHandler: function(e) {
+        this.stopEdgePeek();
+      },
+
+      onClick: function(e) {
+        if (e.target && this.toggleAttribute &&
+            e.target.hasAttribute(this.toggleAttribute)) {
```

Comment:
Maybe assign the result of this boolean logic to a temp that literally describes its meaning?